lib/live_debugger/live_views/channel_dashboard_live.ex

```diff
@@ -187,25 +197,65 @@ defmodule LiveDebugger.LiveViews.ChannelDashboard do
     assign(socket, :node_id, nil)
   end
 
-  defp assign_base_url(socket) do
-    assign(socket, :base_url, "/#{socket.assigns.socket_id}")
-  end
+  defp start_async_assign_lv_process(socket, %{
+         "socket_id" => socket_id,
+         "transport_pid" => transport_pid
+       }) do
+    case Parsers.string_to_pid(transport_pid) do
+      {:ok, pid} ->
+        socket
+        |> assign(:lv_process, AsyncResult.loading())
+        |> start_async(:fetch_lv_process, fn ->
+          fetch_lv_process(socket_id, pid)
+        end)
 
-  defp assign_async_debugged_lv_process(socket) do
-    socket_id = socket.assigns.socket_id
+      :error ->
+        assign(
+          socket,
+          :lv_process,
+          AsyncResult.failed(AsyncResult.loading(), :invalid_transport_pid)
+        )
+    end
+  end
 
+  defp start_async_assign_lv_process(socket, %{"socket_id" => socket_id}) do
     socket
-    |> assign(:debugged_lv_process, AsyncResult.loading())
-    |> start_async(:fetch_debugged_lv_process, fn ->
-      with nil <- fetch_lv_process_after(socket_id, 200),
-           nil <- fetch_lv_process_after(socket_id, 800) do
-        fetch_lv_process_after(socket_id, 1000)
-      end
+    |> assign(:lv_process, AsyncResult.loading())
+    |> start_async(:fetch_lv_process, fn ->
+      fetch_lv_process(socket_id)
     end)
   end
 
-  defp fetch_lv_process_after(socket_id, milliseconds) do
-    Process.sleep(milliseconds)
-    LiveViewDiscoveryService.lv_process(socket_id)
+  defp fetch_lv_process(socket_id, transport_pid \\ nil) do
+    fetch_after = fn milliseconds ->
+      Process.sleep(milliseconds)
+
+      case {transport_pid, LiveViewDiscoveryService.lv_process(socket_id, transport_pid)} do
+        {_, nil} ->
+          nil
+
+        {nil, lv_process} ->
+          {:patch_transport_pid, lv_process}
+
+        {_, lv_process} ->
+          {:noop, lv_process}
+      end
```

Comment:
Couldn't this function return only process struct or nil? In handle async you should be able to check if transport pid is nil. Also, coulnd't we just always trigger patching URL even when nothing has changed? Will it reload the page?